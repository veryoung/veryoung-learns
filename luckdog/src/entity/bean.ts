/** 右上角跳转链接 */
export interface RightLink {
  /** 跳转地址 */
  sUrl: string;
  /** 跳转地址文案 */
  sText: string;
}

/** 跳转更多 */
export interface MoreUrl {
  vText?: string;
}

export interface ItemBean {
  report_info?: Array<Array<string>>;
  ui_style?: number;
  parsedObject?: Record<string, any>;
  item_id?: string;
  tab_id?: string;
  title?: string;
  bottomLineStyle?: number;
  reportState?: boolean;
  symbolKey?: string;
  business?: number;
  url?: string;
}
