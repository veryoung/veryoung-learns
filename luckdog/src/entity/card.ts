/* 卡片左上角链接类型 */
export enum CardLinkType {
  NONE = 0, // 无实验下发(即本地刷新换一换，用已有的书籍进行轮换)
  MORE = 1, // 更多
  CHANGE = 2, // 局部刷新换一换（命中该实验，通过预加载远程书籍换一换）
}


/** 局部刷新换一换请求参数 */
export interface CardReportReq {
  tabId: number, // 当前tab的selectTabID
  tabIdx: number, // 卡片位于tab下的索引
  pageNum: number, // 页码
  refreshKey?: string, // 卡片的refreshKey
}


/** 局部刷新上报参数 */
export interface CardReportInfo {
  [extra: string]: string;
}
