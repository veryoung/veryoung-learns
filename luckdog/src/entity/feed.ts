import { ActiveTab, DynamicTabItem } from '@tencent/luckbox-readdata-feedsentity';

/** 本地缓存动态tab相关数据 */
export interface LocalDynamicTabs {
  dynamicTabs: DynamicTabItem[];
  activeTab: ActiveTab;
}

export * from '@tencent/luckbox-readdata-feedsentity';
