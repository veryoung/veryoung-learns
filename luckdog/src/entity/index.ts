export * from './bean';
export * from './card';
export * from './common';
export * from './feed';
export * from './skin';
export * from './tabs';
