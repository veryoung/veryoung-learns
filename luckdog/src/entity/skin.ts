/**
 * 皮肤的四种模式
 */
export enum SkinModelType {
  /** 正常 */
  NORMAL = 0,
  /** 夜间 */
  NIGHT = 1,
  /** 浅色 */
  LIGHT = 2,
  /** 深色 */
  DARK = 3,
}
