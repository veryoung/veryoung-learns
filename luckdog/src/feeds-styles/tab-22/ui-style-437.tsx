import React from 'react';
import { ScrollView, View, Text, StyleSheet } from '@tencent/hippy-react-qb';

import FeedsProtect from '../../mixins/FeedsProtect';
import FeedsTheme from '../../framework/FeedsTheme';
import { BusiKey, reportUDS, strictExposeReporter } from '@/luckdog';
import { emitter, events } from '../../utils/emitter';
import FeedsUtils from '../../framework/FeedsUtils';
import { TagsBgColors } from '../../framework/FeedsConst';
import { FeedsViewUIStyle437Props } from './__tests__/mocks/ui-style-437';

/** 新协议结构类型 */
type UICard437Type = typeof FeedsViewUIStyle437Props;
type Classify = typeof FeedsViewUIStyle437Props.classifyList[0];

interface CompProps {
  /** 第几刷数据 */
  index: number;
  /** 当前tabid */
  selectTabID: number;
  /** 卡片后台数据 */
  itemBean: {
    parsedObject?: {
      sData: UICard437Type; // 新协议结构
    }
  };
}

/**
 * 新分类导航卡片（不带书籍）
 * 设计稿: https://codesign.woa.com/s/na4Jd0NeeK9AMkb
 */
@FeedsProtect.protect
export default class FeedsViewUIStyle437 extends React.Component<CompProps> {
  private navScrollView: any = null;

  public componentDidMount() {
    emitter.on(events.REFRESH_COMPLETE, this.onTabRefresh);
  }

  public componentWillUnmount() {
    emitter.off(events.REFRESH_COMPLETE, this.onTabRefresh);
  }

  public shouldComponentUpdate(nextProps: CompProps) {
    const { classifyList = [] } = this.props.itemBean.parsedObject?.sData || {};
    const { classifyList: nextClassifyList = [] } = nextProps.itemBean.parsedObject?.sData || {};
    const classifyIds = classifyList.map(item => item?.id).join(',');
    const nextClassifyIds = nextClassifyList.map(item => item?.id).join(',');
    return classifyIds !== nextClassifyIds;
  }

  public render() {
    const { itemBean } = this.props;
    const { classifyList = [] } = itemBean.parsedObject?.sData || {};
    const baseBgIdx = Math.floor(Math.random() * TagsBgColors.length);
    return <ScrollView
      ref={(ref) => {
        this.navScrollView = ref;
      }}
      horizontal
      showsHorizontalScrollIndicator={false}
      scrollEventThrottle={10}
      contentContainerStyle={styles.navWrapper}
      onLayout={this.onLayout}
    >
      {classifyList.map((item, index) => this.renderNavView(item, index, baseBgIdx))}
    </ScrollView>;
  }

  /** 刷新当前tab */
  private onTabRefresh = () => {
    this.navScrollView?.scrollTo({ x: 0, y: 0, animated: true });
  };

  /** 渲染单个导航 */
  private renderNavView = (item: Classify, index: number, baseBgIdx: number) => {
    const { name, jumpUrl } = item;
    return <View
      key={`${name}_${index}`}
      style={[
        styles.navView,
        {
          backgroundColors: TagsBgColors[(baseBgIdx + index) % 5],
          marginLeft: index === 0 ? 0 : 12,
        },
      ]}
      onClick={() => this.jumpClassifyPage(jumpUrl, index)}>
      <Text style={styles.navName}>{name}</Text>
    </View>;
  };

  /** 卡片layout，准备卡片的精准曝光 */
  private onLayout = (event) => {
    strictExposeReporter.addExpoItem({
      cardIndex: this.props.index,
      rect: event.layout,
      forceCheckExpo: true,
    });
  };

  /** 点击分类跳转 */
  private jumpClassifyPage = (jumpUrl: string, index: number) => {
    const { selectTabID, itemBean } = this.props;
    const { reportInfo } = itemBean.parsedObject?.sData || {};
    reportUDS(BusiKey.CLICK__CARD_TAB, {}, {
      ...reportInfo,
      ext_data1: `${index}`,
    }); // 点击上报
    FeedsUtils.doLoadUrl(jumpUrl, `${selectTabID}`);
  };
}

const styles = StyleSheet.create({
  navWrapper: {
    marginTop: 16,
    paddingHorizontal: 16,
  },
  navView: {
    height: 28,
    backgroundColors: FeedsTheme.SkinColor.A8,
    borderRadius: 14,
    paddingHorizontal: 12,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  navName: {
    fontSize: 14,
    colors: FeedsTheme.SkinColor.A1,
    textAlign: 'center',
  },
});
