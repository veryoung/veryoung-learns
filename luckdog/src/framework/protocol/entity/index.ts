
export * from './author';
export * from './banner';
export * from './book-shelf';
export * from './books';
export * from './catch-up';
export * from './category';
export * from './kol';
export * from './nav';
export * from './tag';
export * from './video';
export * from './rank';
export * from './award';
export * from './award-record';
export * from './topic';
