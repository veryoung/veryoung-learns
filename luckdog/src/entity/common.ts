import { View } from '@tencent/hippy-react-qb';
import { ItemBean } from './bean';

export interface CommonProps {
  /** 父元素 */
  parent?: any;
  parents?: any;
  /** 全局变量 */
  globalConf?: any;

  itemBean?: ItemBean;

  index?: number;
  selectTabID?: number;

  /** 组件渲染回调 */
  onLayout?: (event: { layout: Layout }) => void;

  doBeaconByClick?: (...args) => void;

  doBeaconBySlide?: (...args) => void;
}

/** 终端布局 */
export interface Layout {
  /** 相对父元素 x轴上偏向量 */
  x: number,
  /** 相对父元素 y轴上偏向量 */
  y: number,
  /** 宽度 */
  width: number,
  /** 高度 */
  height: number
}

export type ViewRef = View;
