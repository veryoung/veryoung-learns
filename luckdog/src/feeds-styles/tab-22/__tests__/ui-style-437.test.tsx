import { TabId } from '../../../entity';
import React from 'react';
import renderer from 'react-test-renderer';

import FeedsViewUIStyle437 from '../ui-style-437';
import { FeedsViewUIStyle437Props } from './mocks/ui-style-437';

describe('<FeedsViewUIStyle437/>', () => {
  describe('render testing', () => {
    it('render oneRowThreeColumns', () => {
      const props = {
        index: 1,
        selectTabID: TabId.BOTTOM_RECOMM1,
        itemBean: {
          parsedObject: {
            sData: FeedsViewUIStyle437Props,
          },
        },
      };
      const tree = renderer.create(<FeedsViewUIStyle437 {...props} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
