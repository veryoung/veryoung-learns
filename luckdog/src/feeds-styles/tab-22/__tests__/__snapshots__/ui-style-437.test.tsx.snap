// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedsViewUIStyle437/> render testing render oneRowThreeColumns 1`] = `
<div
  contentContainerStyle={
    Object {
      "marginTop": 16,
      "paddingHorizontal": 16,
    }
  }
  horizontal={true}
  nativeName="ScrollView"
  onLayout={[Function]}
  scrollEventThrottle={10}
  showsHorizontalScrollIndicator={false}
  style={
    Object {
      "flexDirection": "row",
      "flexGrow": 1,
      "flexShrink": 1,
      "overflow": "scroll",
    }
  }
>
  <div
    nativeName="View"
    style={
      Array [
        Object {
          "collapse": false,
          "flexDirection": "row",
        },
        Object {
          "marginTop": 16,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(120, 130, 205, 0.1)",
              "rgba(67, 73, 105, 0.1)",
              "rgba(120, 130, 205, 0.1)",
              "rgba(67, 73, 105, 0.1)",
            ],
            "marginLeft": 0,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="全部分类"
      >
        全部分类
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(239, 191, 138, 0.1)",
              "rgba(176, 144, 108, 0.1)",
              "#FDF8F3",
              "#343231",
            ],
            "marginLeft": 12,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="仙侠"
      >
        仙侠
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(255, 121, 110, 0.1)",
              "rgba(121, 69, 67, 0.1)",
              "rgba(255, 121, 110, 0.1)",
              "rgba(121, 69, 67, 0.1)",
            ],
            "marginLeft": 12,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="奇幻"
      >
        奇幻
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(76, 178, 255, 0.1)",
              "rgba(49, 92, 125, 0.1)",
              "rgba(76, 178, 255, 0.1)",
              "rgba(49, 92, 125, 0.1)",
            ],
            "marginLeft": 12,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="悬疑"
      >
        悬疑
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(139, 206, 206, 0.1)",
              "rgba(75, 103, 105, 0.1)",
              "rgba(139, 206, 206, 0.1)",
              "rgba(75, 103, 105, 0.1)",
            ],
            "marginLeft": 12,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="青春"
      >
        青春
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(120, 130, 205, 0.1)",
              "rgba(67, 73, 105, 0.1)",
              "rgba(120, 130, 205, 0.1)",
              "rgba(67, 73, 105, 0.1)",
            ],
            "marginLeft": 12,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="游戏"
      >
        游戏
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(239, 191, 138, 0.1)",
              "rgba(176, 144, 108, 0.1)",
              "#FDF8F3",
              "#343231",
            ],
            "marginLeft": 12,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="现言"
      >
        现言
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(255, 121, 110, 0.1)",
              "rgba(121, 69, 67, 0.1)",
              "rgba(255, 121, 110, 0.1)",
              "rgba(121, 69, 67, 0.1)",
            ],
            "marginLeft": 12,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="重生"
      >
        重生
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(76, 178, 255, 0.1)",
              "rgba(49, 92, 125, 0.1)",
              "rgba(76, 178, 255, 0.1)",
              "rgba(49, 92, 125, 0.1)",
            ],
            "marginLeft": 12,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="风水"
      >
        风水
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(139, 206, 206, 0.1)",
              "rgba(75, 103, 105, 0.1)",
              "rgba(139, 206, 206, 0.1)",
              "rgba(75, 103, 105, 0.1)",
            ],
            "marginLeft": 12,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="女强"
      >
        女强
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(120, 130, 205, 0.1)",
              "rgba(67, 73, 105, 0.1)",
              "rgba(120, 130, 205, 0.1)",
              "rgba(67, 73, 105, 0.1)",
            ],
            "marginLeft": 12,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="都市修仙"
      >
        都市修仙
      </p>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColors": Array [
              "#ffffff",
              "#A6A9B0",
              "#A6A9B0",
              "#A6A9B0",
            ],
            "borderRadius": 14,
            "flexDirection": "column",
            "height": 28,
            "justifyContent": "center",
            "paddingHorizontal": 12,
          },
          Object {
            "backgroundColors": Array [
              "rgba(239, 191, 138, 0.1)",
              "rgba(176, 144, 108, 0.1)",
              "#FDF8F3",
              "#343231",
            ],
            "marginLeft": 12,
          },
        ]
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        text="机甲"
      >
        机甲
      </p>
    </div>
  </div>
</div>
`;
